### DISCUSSION TITLE
TestSuiteContext Before/After hooks with context

### DISCUSSION THREAD

**Discussion by nvcnvn** [cucumber/godog#433] [Ideas]
https://pkg.go.dev/github.com/cucumber/godog#TestSuiteContext
Hi @vearutop , what do you think about this idea:

Given the TestSuiteContext with contextualized hooks
When running a test suite
The context will be properly chained through ScenarioContext hooks
And also chained through the step

---
Use case:
I want to split a huge test suite to smaller one, maybe to many small folder or using multiple tags and run tag by tag.
Having a context from TestSuite level, I can config something like: no more than 10 minutes a run.

**Comment by vearutop** [comment]
Do you mean to start every scenario from a custom `context.Context` of the whole suite?

Or do you mean to pass resulting context of first scenario as starting context of next scenario?

**Reply by nvcnvn** [reply]
> Do you mean to start every scenario from a custom context.Context of the whole suite?

This one sir.

**Reply by vearutop** [reply]
This should be possible now with [`DefaultContext`](https://pkg.go.dev/github.com/cucumber/godog@v0.12.1/internal/flags#Options) option.

Example:
```go
	suite := godog.TestSuite{
...
		Options: &godog.Options{
...
			DefaultContext: myCtx,
		},
	}
```
